src/mqt/ionshuttler/linear/actions.py

```
@@ -0,0 +1,480 @@
+    def __post_init__(self) -> None:
+        """Validate an action's duration when it defines one.
+
+        Raises:
+            ValueError: If the duration is not a positive integer.
+        """
+        duration = getattr(self, "duration", None)
+        if duration is None:
+            return
+        if isinstance(duration, bool) or not isinstance(duration, int):
+            msg = "action duration must be an integer"
+            raise ValueError(msg)  # ruff: ignore[type-check-without-type-error] - Model validation uses ValueError.
```

Ruff (0.16.1)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)